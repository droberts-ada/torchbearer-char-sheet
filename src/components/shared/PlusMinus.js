import React from 'react';
import PropTypes from 'prop-types';

import './styles/PlusMinus.css';

const PlusMinus = function({ name, subtext, value, min, max, onValueChange }) {
  return (
    <li className="control plus-minus">
      <div>
        <h3>{name}</h3>
        <div>
          <button
            onClick={() => onValueChange(value - 1)}
            disabled={value <= min}
            >-</button>
          <span className="number">{value}</span>
          <button
            onClick={() => onValueChange(value + 1)}
            disabled={value >= max}
            >+</button>
        </div>
      </div>
      { subtext ? (<p>{subtext}</p>) : "" }
    </li>
  );
};

PlusMinus.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  onValueChange: PropTypes.func.isRequired,
  subtext: PropTypes.string,
  min: PropTypes.number,
  max: PropTypes.number,
};

export default PlusMinus;
