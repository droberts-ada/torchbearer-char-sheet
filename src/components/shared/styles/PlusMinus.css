.plus-minus > div {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.plus-minus button {
  font-size: 1.4em;
  width: 1.4em;
  text-align: center;
}

.plus-minus .number {
  margin: 0 .5em;
}
