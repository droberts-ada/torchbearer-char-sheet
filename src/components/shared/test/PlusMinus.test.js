import React from 'react';
import renderer from 'react-test-renderer';

import PlusMinus from '../PlusMinus';

it('renders with all props', () => {
  const tree = renderer
  .create(
    <PlusMinus
      name="Test plus minus"
      value={3}
      onValueChange={jest.fn()}
      subtest="test subtext for the component"
      min={0}
      max={4}
      />
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
});

it('disables buttons at boundaries', () => {
  const tree = renderer
  .create(
    <div>
      <PlusMinus
        name="plus minus at min"
        value={0}
        onValueChange={jest.fn()}
        min={0}
        max={4}
        />
      <PlusMinus
        name="plus minus at max"
        value={4}
        onValueChange={jest.fn()}
        min={0}
        max={4}
        />
    </div>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
});
