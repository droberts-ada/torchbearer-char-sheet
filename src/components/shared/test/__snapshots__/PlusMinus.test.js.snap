// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disables buttons at boundaries 1`] = `
<div>
  <li
    className="plus-minus control"
  >
    <div>
      <h3>
        plus minus at min
      </h3>
      <div>
        <button
          disabled={true}
          onClick={[Function]}
        >
          -
        </button>
        <span
          className="number"
        >
          0
        </span>
        <button
          disabled={false}
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </div>
    
  </li>
  <li
    className="plus-minus control"
  >
    <div>
      <h3>
        plus minus at max
      </h3>
      <div>
        <button
          disabled={false}
          onClick={[Function]}
        >
          -
        </button>
        <span
          className="number"
        >
          4
        </span>
        <button
          disabled={true}
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </div>
    
  </li>
</div>
`;

exports[`renders with all props 1`] = `
<li
  className="plus-minus control"
>
  <div>
    <h3>
      Test plus minus
    </h3>
    <div>
      <button
        disabled={false}
        onClick={[Function]}
      >
        -
      </button>
      <span
        className="number"
      >
        3
      </span>
      <button
        disabled={false}
        onClick={[Function]}
      >
        +
      </button>
    </div>
  </div>
  
</li>
`;
